builder/azure/arm/step_publish_to_shared_image_gallery.go

```diff
@@ -56,6 +56,15 @@ func (s *StepPublishToSharedImageGallery) publishToSig(ctx context.Context, mdiI
 	} else {
 		endOfLifeDate = (*date.Time)(nil)
 	}
+
+	var storageAcccountType compute.StorageAccountType
```

Comment:
Is this switch necessary? By my reading, we've already guaranteed that we're using appropriate string constants in the config's Prepare() method

Reply:
This may just be inexperience with Go on my part, but I ended up adding that switch because the `miSigStorageType` parameter in that function is of type string, and the `StorageAccountType` parameter for `compute.GalleryImageVersionPublishingProfile` needs to be of the type `compute.StorageAccountType`, so it's basically just doing the type conversion again.